include/solvers/mpm_base.tcc

```diff
@@ -823,16 +823,35 @@ bool mpm::MPMBase<Tdim>::initialise_math_functions(const Json& math_functions) {
       const std::string function_type =
           function_props["type"].template get<std::string>();
 
+      // Initiate another function_prop to be passed
+      auto function_props_update = function_props;
```

Comment:
why we need to copy here?

Reply:
`function_props` is const. If an input file is passed, the `function_props` must be updated based on the file prior to creating the math function. Thus, mutable copy is used.